### PR TITLE
dep(dm): move column-mapping from tidb-tools to tiflow repo

### LINKED ISSUE #11114: move column-mapping from tidb-tools to tiflow repo

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #11114

### What is changed and how it works?

column-mapping is a dead package, we won't develop its feature in future, so the duplication of code with tidb-tools is fine because the code will not change.

#### Tests 

 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by 3AceShowHand** [comment]
please create new issue for each PR, this makes the cherry-pick process easier.

**Review by 3AceShowHand — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

File: pkg/column-mapping/column.go
Comment (D3Hunter) [inline pkg/column-mapping/column.go:1]: are all code copied from tidb repo?

Comment (lance6716) [reply pkg/column-mapping/column.go:1]: tidb-tools. And the test files are changed to `require` package

**Pull request merged**